### ISSUE TITLE
Daggerhashimoto not using all gpus (2x gtx 970 + 2x gtx 1070)

### ISSUE THREAD

**Issue by esensio** [nicehash/NiceHashMiner-Archived#187]
Hi,
I am using the latest version of nicehash 1.6.1.4 on Win7x64 with nvidia driver 372.54.
I have 4 GPUs: 2 x GTX 970 + 2 x GTX 1070
When use to mine with daggerhashimoto only 1 gpu from every group is mining.
I need to put in extra options "--cuda-devices 0 1 2 3" to use all 4 gpu's. The total hashing power is ~90MH/s for all 4 gpus. I think the 1070s do not hash as expected.

Any ideas how to fix this?

Hashing Speed - Daggerhashimoto /per card
GTX 970 ~20MH/s +-10%
GTX 1070 ~25MH/s (not good! ) --> I have ~30MH/s with genoil's 1.1.7 which is still slow, but I hope to be fixed in future releases.

GPUs
1x ASUS STRIX ROG GTX 1070 8GB OC
1x GIGABYTE GTX 1070 G1 Gaming OC 8GB
2x PNY GTX 970 4GB OC XLR8
MB/CPU/RAM
GA-Z97X-SOC / G1840 / 8GB

**Comment by esensio** [comment]
[log.txt](https://github.com/nicehash/NiceHashMiner/files/438122/log.txt)

**Comment by S74nk0** [comment]
Hi @esensio. NiceHashMiner_v1.6.1.4 does not support officially SM6.x cards and Daggerhashimoto is disabled (although you have provided a workaround). Try our public beta https://github.com/nicehash/NiceHashMiner/releases/tag/v1.7.0.1-BETA. This pre-release officially supports SM6.x cards. If you encounter any problems with the BETA release please open a new issue and we will fix it. We will take a look into ethminer speed problem.

**Comment by esensio** [comment]
Thank you Stanko :) I am already running the latest beta. Dagger runs fine, although the ethminer version is the same. There's no change in hashing speed of 1070's.
So far I tested Genoil's, Claymore's and Nicehash forks of ethminer both 6.5 and 8rc releases and branches 1.0.x and 1.1.x and it seems they all report the same hashing speed of 1070's - about 25-26MH/s. 
Sometimes there's a boost of 10MH/s which can happen in the beginning of mining for a few seconds, when the gpus are colder and the clocks are reaching ~2000MHz, but even at 1936MHz on 52 degrees Celsius the hashing speed is still about 25MH/s using dagger.
I have installed both .NET 4.5.2 and CUDA 8 RC, just to check if there will be any boost in performance (with cuda install), but as I was certainly sure, there's no such :)

Attaching two screenshots from cmd version of "ethminer-0.9.41-genoil-1.1.7" (not 6.5) configured for mining on nanopool and nicehash with these commands:

1) for Nanopool: "ethminer.exe --farm-recheck 200 -U -S eth-eu1.nanopool.org:9999 -FS eu2.nanopool.org:9999 -O myETHwalletaddress.SOC-ETH/myemail --cuda-grid-size 8192 --cuda-block-size 128"
![ethminer-117-nanopool](https://cloud.githubusercontent.com/assets/21242805/18202526/304bb7fa-7119-11e6-9ebb-3faa6fa92bb9.jpg)

2) for Nicehash: "ethminer -U -S daggerhashimoto.eu.nicehash.com:3353 -O myBTCAddress.SOC:x --stratum-protocol 2 --cuda-grid-size 8192 --cuda-block-size 128"
![ethminer-117-nicehash](https://cloud.githubusercontent.com/assets/21242805/18202529/31fc5b5e-7119-11e6-9532-a47c163f16b8.jpg)
These are my experiences so far. I hope to be helpful for you.

Regards.
